### PR TITLE
Upgrade kubernetes dependency to v1.19.14

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Update kubernetes dependency to v1.19.14

### What is changed and how does it work?

1. exec `hack/pin-deps.sh 1.19.14`
2. fix some dependency manually, contain:
    * upgrade github.com/prometheus/client_golang v0.9.4 to v1.11.0
    * upgrade sigs.k8s.io/controller-runtime v0.4.0 to v0.7.2
    * use go.etcd.io/etcd to replace github.com/coreos/etcd
    * downgrade github.com/googleapis/gnostic to v0.4.1
3. all code where call kubenetes client add `context.Context`
4. modify `GenericControlInterface` to be compatible with controll-runtime `client.Object`
5. modify e2e to be compatible with kubenetes e2e

### Code changes

- [x] Has Go code change
- [x] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 

Kind 1.12.5
- [x] TiDBCluster
- [x] Initializer
- [x] TiDBMonitor
- [x] DM
- [x] Backup Restore

EKS 1.16.15
- [x] TiDBCluster
- [x] Initializer
- [x] TiDBMonitor
- [x] DM
- [x] Backup Restore

EKS 1.17.16
- [x] TiDBCluster
- [x] Initializer
- [x] TiDBMonitor
- [x] DM
- [x] Backup Restore

EKS 1.19.13
- [x] TiDBCluster
- [x] Initializer
- [x] TiDBMonitor
- [x] DM
- [x] Backup Restore 

EKS 1.20.7
- [x] TiDBCluster
- [x] Initializer
- [x] TiDBMonitor
- [x] DM
- [x] Backup Restore 

EKS 1.21
- [x] TiDBCluster
- [x] Initializer
- [x] TiDBMonitor
- [x] DM
- [x] Backup Restore

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Update kubernetes dependency to v1.19.14 
```

### CODE REVIEW

**Review by KanShiori — COMMENTED** [review]
Ref: https://github.com/kubernetes/kubernetes/pull/88540

File: go.mod [hunk 1]
```diff
@@ -28,73 +24,68 @@ require (
...
+	github.com/spf13/pflag v1.0.5
 	github.com/tikv/pd v2.1.17+incompatible
-	github.com/tmc/grpc-websocket-proxy v0.0.0-20171017195756-830351dc03c6 // indirect
 	github.com/uber-go/atomic v0.0.0-00010101000000-000000000000 // indirect
 	github.com/uber/jaeger-client-go v2.19.0+incompatible // indirect
 	github.com/uber/jaeger-lib v2.2.0+incompatible // indirect
 	github.com/xi2/xz v0.0.0-20171230120015-48954b6210f8 // indirect
-	github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2 // indirect
 	github.com/yisaer/crd-validation v0.0.3
+	go.etcd.io/etcd v0.5.0-alpha.5.0.20200819165624-17cef6e3e9d5
```
Comment (KanShiori) [inline go.mod:66]: the version is same as the version in kubenetes's go.mod

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -28,8 +29,7 @@ import (
 	eventv1 "k8s.io/client-go/kubernetes/typed/core/v1"
 	"k8s.io/client-go/tools/record"
 	"k8s.io/klog"
-	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
-	schedulerapiv1 "k8s.io/kubernetes/pkg/scheduler/api/v1"
+	schedulerapi "k8s.io/kube-scheduler/extender/v1"
```
Comment (KanShiori) [inline pkg/scheduler/scheduler.go:32]: ref https://github.com/kubernetes/kubernetes/pull/88540

File: tests/e2e/e2e_test.go [hunk 3]
```diff
@@ -119,14 +119,6 @@ func TestMain(m *testing.M) {
 		log.Logf("FLAG: --%s=%q", flag.Name, flag.Value)
 	})
 
-	// Now that we know which Viper config (if any) was chosen,
```
Comment (KanShiori) [inline tests/e2e/e2e_test.go:122]: it's useless, ref: https://github.com/kubernetes/kubernetes/pull/102598

File: go.mod [hunk 4]
```diff
@@ -28,73 +24,68 @@ require (
...
-	github.com/onsi/ginkgo v1.10.3
-	github.com/onsi/gomega v1.5.0
-	github.com/openshift/generic-admission-server v1.14.0
+	github.com/onsi/ginkgo v1.14.1
+	github.com/onsi/gomega v1.10.2
+	github.com/openshift/generic-admission-server v1.14.1-0.20210422140326-da96454c926d
 	github.com/opentracing/opentracing-go v1.1.0 // indirect
 	github.com/pierrec/lz4 v2.0.5+incompatible // indirect
-	github.com/pingcap/advanced-statefulset/client v1.16.0
+	github.com/pingcap/advanced-statefulset/client v1.17.1-0.20210831081013-d54ef54b2938
```
Comment (KanShiori) [inline go.mod:44]: master branch in advanced-statefulset

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**